src/lib/editable-feature-collection.js

```diff
@@ -200,54 +209,124 @@ export class EditableFeatureCollection {
   /**
    * Returns a flat array of positions for the given feature along with their indexes into the feature's geometry's coordinates.
    *
-   * @param featureIndex The index of the feature to get edit handles
+   * @param featureIndexes The indexes of the features to get edit handles
+   * @param project Projection method to convert geographic coordinates to screen coordinates
+   * @param pointerPosition Object containg screen and ground coordinates for the location of the pointer
    */
-  getEditHandles(featureIndex: number) {
+  getEditHandles(featureIndexes: Array<number>, project?: Function, pointerPosition?: Object) {
```

Comment:
I think instead of getting a `project` function, screen coordinates, and ground coordinates, this should instead just get an array of picked features and a ground coordinate (no screen coordinates). The picked features would be the ones that are within deck.gl's picking threshold. Then `getEditHandles` only needs to look through those picks to see if any of them are lines or polygons in order to get the closest point along the lines. It can also short circuit the nearestPointOnLine check if one of those picks is an existing edit handle.

One benefit when someone has lots of selections. Consider someone has hundreds of features loaded and all are selected. In that case, you'd have to run `nearestPointOnLine` on thousands or millions of lines.

Another benefit is that the pixel picking threshold is in one place, deck.gl. This would simplify the picking logic and make it consistent. I noticed a slight discrepancy when dragging close enough to get the intermediate point to appear but far enough away that deck.gl doesn't consider the intermediate point clicked; in that case, the map pans rather than a point getting added.

Here's how you could achieve that. Move the call to `getEditHandles` from `EditableGeoJsonLayer.updateState` to `EditableGeoJsonLayer.onPointerMove`. And in `onPointerMove` do something like this:
```
const picks = this.context.layerManager.pickObject({
  x: screenCoords[0],
  y: screenCoords[1],
  mode: 'query',
  layers: [this.props.id],
  radius: 10,
  viewports: [this.context.viewport]
});
editableFeatureCollection.getEditHandles(props.selectedFeatureIndexes, picks, groundCoords);
```